sympy/stats/tests/test_rv.py

```diff
@@ -227,3 +227,7 @@ def test_issue_8129():
     assert P(X >= X) == 1
     assert P(X > X) == 0
     assert P(X > X+1) == 0
+
+def test_issue_12237():
+    X = Normal('X', 0, 1)
+    assert P(X > 0, X) == 1/2
```

Comment:
isn't this supposed to be a function?

Reply:
I mean, I would expect `P(A, B)` to create a `RandomSymbol` or similar.

Reply:
I cannot get your question. 
This test is passing, so any other issues with this? 
 Thanks for all of the reviews and discussion. :-)

Reply:
Oh I see, @Upabjojr  this wasn't visible to me a few minutes back, got it.
> I mean, I would expect `P(A, B)` to create a `RandomSymbol` or similar.
So, is this the accurate requirement of the issue as well?

Reply:
Well, supporting random variables in the condition of `P` is much more complicated than your solution in this PR. `P(X>0, X)` is not a number, it's a random variable itself. This result is definitely wrong.

Reply:
@czgdp1807 can you determine which random variable is `P(X>0, X)` equivalent to, in this case?

Reply:
If `Z = P(X>0, X=z)` and `Z` is a random variable then, according to my point of view, `Z` follows the distribution such that when, `z > 0`, the probability is 1 and when `z <= 0` then probability is 0.
A short explanation:
If z > 0, then we are given that X has some positive value, so `P(X > 0, X = z)` will be 1 and if z < 0, then we are given that X > 0 is not possible, so,`P(X > 0, X = z)` will be 0. We can also say that it's [discrete unit step function](https://en.wikipedia.org/wiki/Heaviside_step_function#Discrete_form).
Please correct if I am wrong.

Reply:
I see it like this. A random variable maps events to real numbers. The density function it associates to real numbers it the probability of the events. So `P(X>0|X)` maps event `E` to `P(X>0|X = X(E))`. If `X(E) > 0`, which happens 1/2 of the times, then the value of `P(X>0)` is 1, otherwise it's 0.

I would say that `P(X>0|X)` is a Bernoulli random variable with `p=1/2` when `X` is normal centered at the origin.

Reply:
I think your code is returning a completely wrong result.

Reply:
Yes @Upabjojr your opinion is right. Infact, `P(Z = 1) = 1/2` since P(X > 0) is 1/2. Moreover, I think rather than saying probability is 1, I should have said value of Z is 1.
> when z >0, the probability is 1 and when `z <= 0` then probability is 0.  
Hence, P(Z = 1) whenever X > 0, and P(X > 0) is 1/2 so, P(Z = 1) = 1/2. I totally agree with your opinion. Thanks for the correction. :-)
Agreed on you second comment too. I am working on some more tests as suggested by you. I will write the code after that.

Reply:
There is a `Probability` class, it's meant for unevaluated probability expressions. I would suggest to start with that one.

That is, `P(..., X)` should return `Probability(..., X)` without evaluating the expression. In the future more specific algorithms may be added.

Reply:
@Upabjojr I have made some changes. I think this level of abstraction doesn't need more tests. 
I believe this partially solves the issue. However, more work is to be done for returning a random variable for P(A|X).

Reply:
There is a potential problem that I see. Now `Probability` may be both a random and non-random expression, depending on the `given_condition` object. We need to devise some API change to support this.

Reply:
I would suggest that SymPy objects be equipped with an `.is_random` property, defaulting to `False`.

Reply:
But maybe this is for a future PR.

Reply:
@Upabjojr I agree, that this will create a problem. However, I think rather than giving, `is_random` attribute to all the sympy objects, instead, we can give, `is_random` to only `Probability` class and we can set the attribute to `True` according to the situation. This approach will require lesser and manageable changes and will not affect the code written earlier, as far as I can see.

Example:
```
>>> X = Normal('X', 0, 1)
>>> Z = P(X > 0, X > 0)
>>> Z.is_random
False
>>> Z = P(X > 0, X)
>>> Z.is_random
True
```